### DISCUSSION TITLE
Bubbling Guitar and Clear Speech from same set-up

### DISCUSSION THREAD

**Discussion by TPennick** [jamulussoftware/jamulus#1465] [Technical Support & Help]
Hi All,
I have a private server on a Linode Ubuntu virtual server, (the smallest "Nanode" option).  When I connect to the server with no other participants, I get a very clear return sound with no bubbles or crackles and very good latency (total about 24ms).  However, one of my band, connecting via Windows and a Focusrite Scarlett Solo, produces very clear speech from his mic, but very bubbly sounding guitar plugged into the same interface.  With 4 members of the band connected, I also get some slight bubbling on my instrument, which is a mic'd accordion.  I'm wondering why the guitarist should find that his mic is unaffected by bubbling, where the sound quality of the guitar is quite poor for a proportion of the time (it seems to get better as the session proceeds).  I'm also wondering if the Linode Nanode is likely to be under powered for a band of 4, each having a mic and a separate instrument plugged in.  Other band members, one playing a ukulele through a behringer Um2 connected to a Mac, and the other using another Scarlett Solo through Windows, are achieving very high quality results.
Regards and thanks for any thoughts.

**Comment by dcorson-ticino-com** [comment]
What you say points to client connection problems that may or may not have anything to do with your server.
As you have two members without sound problems the server is most probably not the culprit.

The "bubbly" sounding stuff is caused by data packets arriving out of order.  Jamulus already has automatic packet reordering, but to a limited extent so as not to add to the delay.  That already helps a lot.
Your band members with the sound problems should try changing the buffer settings and maybe turning off the automatic jitter buffer and pushing the jitter buffer size up step by step to see if that helps.

I have no idea about Linode servers, but you could try going to the next bigger offer for a test just to see if maybe it also has better Internet connection properties.
You could also try using another public server in your area that may have better connections to your band members ISPs.  You never know.
A last possibility is the people with problems could try changing their ISP, although that is not always possible, I know.
Such connection problems are always difficult to diagnose and correct.

**Reply by gene96817** [reply]
@dcorson-ticino-com  would you have an idea about:

> produces very clear speech from his mic, but very bubbly sounding guitar plugged into the same interface

Does the codec produce a continuous stream for speech or is the stream intermittent?

**Reply by dcorson-ticino-com** [reply]
I have no really good idea, maybe the characteristics of the waveforms of speech and the guitar make the "bubbling" more or less pronounced.  In any case if the mic and the guitar are being input through the same interface and connection they will be seeing the same distortions.  
As far as I know the codec produces a continuous stream without interruptions.
The biggest problem diagnosing stuff like this is that the Internet is a moving target.  You never know how the routing will be from one day to the next or from one minute to the next.  I have seen it often enough that simply disconnecting and reconnecting a minute later gives a better (or worse) connection.
My orchestra has now been playing for months using my server, but once, one of our members had a really bad connection that stayed that way.  We then all went to another server which worked.  That was 2 months ago, ever since then the service from my server has been fine.  Don't ask me why it was so bad just this one time.  That is the joy of modern technology.
Unfortunately we as users of Jamulus have an unusual, for the Internet, usage and delay, jitter, etc. that makes the connection useless for us will remain completely unseen for 99.99% of users.
In any case if we hear "bubbling" we know that it is an Internet connection quality problem.  Fortunately we are hearing much less about bubbling since the packet reordering code was added, it works, but there are limits to what we can do without adding latency.
